### PR TITLE
Enable dark theme

### LINKED ISSUE #1584: Enable darkmode
We would like to enable the dark mode theme. 
`pi-ui` already supports dark mode which need to be tested and integrated within `pigui`

### PR DESCRIPTION
**Pull request by amassarwi**
This PR adds dark theme feature.
 As specified in provided [specs](https://xd.adobe.com/view/1fea4fa1-a829-428c-48d9-3f72193f1be9-69db/screen/d6aa1833-993b-4514-affc-f80bf64ba081/Proposal-details-Desktop-1/).
Users could switch to dark mode by toggling the new toggle switch which was added as part of this PR to the user's header menu.

Closes #1584

### CODE REVIEW

**Comment by amassarwi** [comment]
@fernandoabolafio
Two minor that should be addressed:
 -  What should we use for dropdown's background color ? I've tried `#1F2F5A` & `#223767` but didn't look good.
 - Couldn't extract the dark theme icon as .svg file from the provided specs ? (Is there a way? :thinking:) - Could you please send to me @MariaPleshkova ? 

image

**Comment by amassarwi** [comment]
@fernandoabolafio @MariaPleshkova
1. We could improve the **header** of the following table - what would be a proper color for that case ?
image
I'll update the separator to the dark one we use in other parts.

2. Notification background could be updated as well:
image

File: src/componentsv2/ModalDiff/ModalDiff.jsx [hunk 1]
```diff
@@ -55,7 +57,7 @@ const ModalDiff = ({
             {proposalDetails.version && (
               <Text
                 id={`proposal-${proposalDetails.proposalToken}-version`}
-                color="gray"
+                color={ isDarkTheme ? "primaryDark" : "gray" }
```
Comment (tiagoalvesdulce) [inline src/componentsv2/ModalDiff/ModalDiff.jsx]: We shouldn't do this kind of conditional in the JSX code. Please create a variable for it so you can use it everywhere in the codebase.

Comment (amassarwi) [reply src/componentsv2/ModalDiff/ModalDiff.jsx]: @tiagoalvesdulce Thanks for the review )
Sure, will do.

Comment (amassarwi) [reply src/componentsv2/ModalDiff/ModalDiff.jsx]: @tiagoalvesdulce updated

Comment (tiagoalvesdulce) [reply src/componentsv2/ModalDiff/ModalDiff.jsx]: way better, thanks!

**Comment by amassarwi** [comment]
@fernandoabolafio 
Thanks for the review, great notes, will be addressed.

**Comment by MariaPleshkova** [comment]
@amassarwi 
1.
> What should we use for dropdown's background color ? I've tried #1F2F5A & #223767 but didn't look good.

As for me, #223767 looks ok as dropdown's background color (I tried lighter one from the palette just in case, but #223767 looks better). I added an example to the spec, so now you can check it there as well.

2. Here's the dark theme icon: 
[Politeia_logo_darkmode.zip](https://github.com/decred/politeiagui/files/3949969/Politeia_logo_darkmode.zip)

3. 

> We could improve the header of the following table - what would be a proper color for that case ?

I would go with #1F2F5A 

4. 

> Notification background could be updated as well:

I think #152042 is ok. As it's a notification, so it has to stand out.

**Comment by amassarwi** [comment]
@MariaPleshkova thanks for your input

File: src/componentsv2/RecordWrapper/RecordWrapper.jsx [hunk 2]
```diff
@@ -26,17 +26,18 @@ export const Author = ({ username, id }) => (
   <Link to={`/user/${id}`}>{username}</Link>
 );
 
-export const Event = ({ event, timestamp }) => (
+export const Event = ({ event, timestamp }) => {
+  return (
```
Comment (amassarwi) [inline src/componentsv2/RecordWrapper/RecordWrapper.jsx]: Redundant return

File: src/componentsv2/Join/Join.jsx [hunk 3]
```diff
@@ -3,9 +3,11 @@ import PropTypes from "prop-types";
 import { classNames } from "pi-ui";
 import styles from "./Join.module.css";
 
-const DefaultSeparator = () => (
-  <span className="margin-left-s margin-right-s color-gray">•</span>
-);
+const DefaultSeparator = () => {
+  return (<span className="text-secondary-color margin-left-s margin-right-s">
```
Comment (amassarwi) [inline src/componentsv2/Join/Join.jsx]: redundant return

**Comment by amassarwi** [comment]
@fernandoabolafio @MariaPleshkova all comments tackled & addressed.

**Comment by MariaPleshkova** [comment]
@amassarwi @fernandoabolafio added new issues to politeiagui

**Pull request merged**